### PR TITLE
Clean-up source tree

### PR DESCRIPTION
**Pull request by gratian**
Reorganize the source tree in preparation for adding more tests:

  * Add a .gitignore file to reduce noise in the git status output from intermediate build files.

  * Split the source and tests into separate folders.

  * Simplify the Makefile.am for tests by taking advantage of automake's default sources[1] behaviour.

[1] https://www.gnu.org/software/automake/manual/html_node/Default-_005fSOURCES.html

Signed-off-by: Gratian Crisan \<gratian.crisan@ni.com\>

### CODE REVIEW

**Review by dvhart — APPROVED** [review]
For the .gitexclude change, I typically only exclude build artifacts, but including the patch and editor artifacts makes sense. Good improvement, thanks.

**Review by dvhart — APPROVED** [review]
The Makefile and source re-org looks fine. Might have preferred to separate the libtool change as that is independent of the organization, but no need to resubmit for that.

File: tests/Makefile.am [hunk 1]
```diff
@@ -2,17 +2,7 @@
 # Copyright © 2018 VMware, Inc. All Rights Reserved.
 
 AM_CPPFLAGS = -I. -I$(top_srcdir)/src
-
-bin_PROGRAMS = test_api tst-condpi2 tst-cond1
-
-test_api_SOURCES = test_api.c rtpi.h
```
Comment (dvhart) [inline tests/Makefile.am]: By removing all references to rtpi.h, and not adding it as an explicit dependency for the test targets - doesn't this break triggering a rebuild when nothing but rtpi.h changes?

Comment (gratian) [reply tests/Makefile.am]: I can confirm experimentally that any change in rtpi.h will trigger a rebuild when running the 'make test' (i.e. 'make check') target.

Would that be a sufficient test? I'm attempting to map out the dependency chain that leads to the rebuild but automake generated makefiles or not the easiest read.

**Comment by dvhart** [comment]
Thanks for confirming the dependency chain for the tests is maintained.

**Pull request merged**